extprocconfig/extprocconfig.go

```diff
@@ -38,24 +47,24 @@ import (
 //	    outputSchema:
 //	      schema: AWSBedrock
 //	  headers:
-//	  - name: x-model-name
+//	  - name: x-envoy-ai-gateway-llm-model
 //	    value: llama3.3333
 //	- backends:
 //	  - name: openai
 //	    outputSchema:
 //	      schema: OpenAI
 //	  headers:
-//	  - name: x-model-name
+//	  - name: x-envoy-ai-gateway-llm-model
 //	    value: gpt4.4444
 //
-// where the input of the external processor is in the OpenAI schema, the model name is populated in the header x-model-name,
-// The model name header `x-model-name` is used in the header matching to make the routing decision. **After** the routing decision is made,
-// the selected backend name is populated in the header `x-backend-name`. For example, when the model name is `llama3.3333`,
+// where the input of the external processor is in the OpenAI schema, the model name is populated in the header x-envoy-ai-gateway-llm-model,
```

Comment:
Since user can configure the model routing header name, we can say configured modelNameHeaderKey

Reply:
well, no, users are not required to choose the model routing header name at this point. https://github.com/envoyproxy/ai-gateway/blob/cb6b2e0f12fb6bc2e23941eff6039cf78d5693ca/api/v1alpha1/api.go#L206

It should be good to make this part of the LLMRoute resource, but the comment here matches what it is now